test/util.js

```diff
@@ -37,8 +37,14 @@ export var assertEqual = function(a, b){
   b.fork(function(x){bstate = 1; bval = x}, function(x){bstate = 2; bval = x});
   if(astate === 0){ throw new Error('First Future passed to assertEqual did not resolve instantly') }
   if(bstate === 0){ throw new Error('Second Future passed to assertEqual did not resolve instantly') }
+  if(isFuture(aval) && isFuture(bval)) return assertEqual(aval, bval);
```

Comment:
This line is the significant change in this file. It allows for "deep" equality of nested Futures. The only limitation being (of course) that the Futures be synchronous.